src/gc/gc_manager.cpp

```diff
@@ -11,51 +11,10 @@
 //===----------------------------------------------------------------------===//
 #include "gc/gc_manager.h"
 
-#include "catalog/schema.h"
-#include "type/types.h"
-#include "type/value.h"
-#include "type/varlen_pool.h"
-#include "storage/tile.h"
-#include "storage/tile_group.h"
-
 namespace peloton {
 namespace gc {
 
-// Check a tuple and reclaim all varlen field
-void GCManager::CheckAndReclaimVarlenColumns(storage::TileGroup *tg, oid_t tuple_id) {
-  oid_t tile_count = tg->tile_count;
-  oid_t tile_col_count;
-  type::Type::TypeId type_id;
-  char *tuple_location;
-  char *field_location;
-  char *varlen_ptr;
-
-  for (oid_t tile_itr = 0; tile_itr < tile_count; tile_itr++) {
-    const catalog::Schema &schema = tg->tile_schemas[tile_itr];
-
-    tile_col_count = schema.GetColumnCount();
 
-    storage::Tile *tile = tg->GetTile(tile_itr);
-    PL_ASSERT(tile);
-    for (oid_t tile_col_itr = 0; tile_col_itr < tile_col_count; ++tile_col_itr) {
-      type_id = schema.GetType(tile_col_itr);
-
-      if ((type_id != type::Type::TypeId::VARCHAR && type_id != type::Type::TypeId::VARBINARY)
-          || (schema.IsInlined(tile_col_itr) == true)) {
-        // Not of varlen type, or is inlined, skip
-        continue;
-      }
-      // Get the raw varlen pointer
-      tuple_location = tile->GetTupleLocation(tuple_id);
-      field_location = tuple_location + schema.GetOffset(tile_col_itr);
```

Comment:
We definitely need this piece of code after we stop hacking the varlen pool because it help the gc manager reclaim memory from the varlen pool. It seems that the "ephemeral" pool does not free memory at all.